packages/@glimmer/runtime/lib/compiled/opcodes/component.ts

```diff
@@ -716,9 +723,29 @@ APPEND_OPCODES.add(Op.PopulateLayout, (vm, { op1: _state }) => {
 });
 
 APPEND_OPCODES.add(Op.VirtualRootScope, (vm, { op1: _state }) => {
-  let { symbols } = check(vm.fetchValue(_state), CheckFinishedComponentInstance).table;
+  let { table, manager, capabilities, state } = check(
+    vm.fetchValue(_state),
+    CheckFinishedComponentInstance
+  );
+
+  let owner;
+
+  if (managerHasCapability(manager, capabilities, InternalComponentCapability.HasSubOwner)) {
+    owner = manager.getOwner(state);
+    vm.loadValue($t1, null); // Clear the temp register
+  } else {
+    // Check the temp register to see if an owner was resolved from currying
+    owner = vm.fetchValue<Owner | null>($t1);
+
+    if (owner === null) {
+      // If an owner wasn't found, default to using the current owner
+      owner = vm.getOwner();
+    } else {
+      vm.loadValue($t1, null); // Clear the temp register
+    }
```

Comment:
Can you add a small example above each of these branches to indicate what thing the user would have done to end up in each case?